### PR TITLE
delivery skip_gc_vars attr to cinn subgraph

### PR DESCRIPTION
**Pull request by CtfGo**
### PR types

Bug fixes
### PR changes

OPs
### Describe

delivery skip_gc_vars from the main graph to each subgraph compiled by CINN

### CODE REVIEW

File: paddle/fluid/framework/paddle2cinn/build_cinn_pass.cc [hunk 1]
```diff
@@ -693,10 +693,27 @@ void SearchAllSubgraphs(Graph* graph, bool is_inference_stage) {
...
+      const auto& all_skip_gc_vars =
+          graph->Get<std::unordered_set<std::string>>(kSkipGcVarNames);
+      const auto& sub_input_vars =
+          subgraph->Get<std::vector<std::string>>(kInputVars);
+      const auto& sub_output_vars =
+          subgraph->Get<std::vector<std::string>>(kOutputVars);
+      auto& sub_skip_gc_vars =
+          subgraph->GetOrInit<std::unordered_set<std::string>>(kSkipGcVarNames);
+      for (auto&& var_name : sub_input_vars) {
+        if (all_skip_gc_vars.count(var_name)) sub_skip_gc_vars.insert(var_name);
```
Comment (Aurelius84) [inline paddle/fluid/framework/paddle2cinn/build_cinn_pass.cc]: 这里不需要用input 和output做过滤

File: paddle/fluid/operators/cinn/cinn_launch_context.cc [hunk 2]
```diff
@@ -94,11 +95,18 @@ CinnLaunchContext::CinnLaunchContext(const framework::ir::Graph& graph,
   auto& outer_varinfo = graph.Get<Name2VarInfoMap>(kMemOptVarInfoFromMainGraph);
   runtime_graph_->SetNotOwned<Name2VarInfoMap>(kMemOptVarInfoFromMainGraph,
                                                &outer_varinfo);
-  // collect skip_eager_vars
+  // use kSkipGcVarNames attr of graph to initialize skip_gc_vars_
+  if (graph.Has(kSkipGcVarNames)) {
+    const auto& skip_gc_vars =
+        graph.Get<std::unordered_set<std::string>>(kSkipGcVarNames);
+    skip_gc_vars_.insert(skip_gc_vars.begin(), skip_gc_vars.end());
```
Comment (Aurelius84) [inline paddle/fluid/operators/cinn/cinn_launch_context.cc:103]: 这里是否可以用string::join_strings +VLOG提供下调试信息？

File: paddle/fluid/framework/paddle2cinn/build_cinn_pass.cc (hunk 1 above)
Comment (CtfGo) [reply paddle/fluid/framework/paddle2cinn/build_cinn_pass.cc]: fix

File: paddle/fluid/operators/cinn/cinn_launch_context.cc (hunk 2 above)
Comment (CtfGo) [reply paddle/fluid/operators/cinn/cinn_launch_context.cc:103]: done

**Review by Aurelius84 — APPROVED** [review]
LGTM

**Pull request merged**